File: src/test/java/com/raychenon/leetcode/KClosestPointsToOriginTest.java
package com.raychenon.leetcode;

import org.junit.Assert;
import org.junit.BeforeClass;
import org.junit.Test;

/**
 * User: raychenon
 * Date: 31/5/2020
 * https://leetcode.com/problems/k-closest-points-to-origin
 */
public class KClosestPointsToOriginTest {

    private static KClosestPointsToOrigin kClosestPointsToOrigin;

    @BeforeClass
    public static void setUp() {
        kClosestPointsToOrigin = new KClosestPointsToOrigin();
    }


    @Test
    public void example1Test() {
        evaluatekClosest(
                new int[][]{{-2, 2}},
                new int[][]{{1, 3}, {-2, 2}}, 1
        );
    }

    @Test
    public void example2Test() {
        evaluatekClosest(
                new int[][]{{3, 3}, {-2, 4}},
                new int[][]{{3, 3}, {5, -1}, {-2, 4}}, 2
        );
    }

    @Test
    public void exampleSameDistanceTest() {
        evaluatekClosest(
                new int[][]{{1, 1}, {-1, 1}},
                new int[][]{{1, 1}, {-1, 1}, {3, 3}, {5, -1}, {-2, 4}}, 2
        );
    }

    void evaluatekClosest(int[][] expected, int[][] points, int K) {
        Assert.assertArrayEquals(
                expected,
                kClosestPointsToOrigin.kClosestSort(points, K)
        );
        Assert.assertArrayEquals(
                expected,
                kClosestPointsToOrigin.kClosestQueue(points, K)
        );
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/KClosestPointsToOriginTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 30/5/2020
 * https://leetcode.com/problems/k-closest-points-to-origin/
 */
class KClosestPointsToOriginTest {

    @Test
    fun example1Test() {
        evaluatekClosest(
            arrayOf(intArrayOf(-2, 2)),
            arrayOf(intArrayOf(1, 3), intArrayOf(-2, 2)), 1
        )
    }

    @Test
    fun example2Test() {
        evaluatekClosest(
            arrayOf(intArrayOf(3, 3), intArrayOf(-2, 4)),
            arrayOf(intArrayOf(3, 3), intArrayOf(5, -1), intArrayOf(-2, 4)), 2
        )
    }

    @Test
    fun exampleSameDistanceTest() {
        evaluatekClosest(
            arrayOf(intArrayOf(1, 1), intArrayOf(-1, 1)),
            arrayOf(
                intArrayOf(1, 1),
                intArrayOf(3, 3),
                intArrayOf(5, -1),
                intArrayOf(-1, 1),
                intArrayOf(-2, 4)
            ), 2
        )
    }

    fun evaluatekClosest(expected: Array<IntArray>, points: Array<IntArray>, K: Int): Unit {
        Assert.assertArrayEquals(
            expected,
            KClosestPointsToOrigin.kClosestOriginal(points, K)
        )
        Assert.assertArrayEquals(
            expected,
            KClosestPointsToOrigin.kClosest(points, K)
        )

    }


}